src/searcher/mod.rs

```diff
@@ -349,11 +351,62 @@ impl Searcher {
         Ok(())
     }
 
+    pub fn search_slice_callback_no_path(
+        &mut self,
+        query_context: QueryContext,
+        slice: RopeOrSlice,
+        // slice: impl TextProvider<'a> + Parseable + 'a,
+        tree: Option<&Tree>,
+        mut callback: impl FnMut(CaptureInfo),
+    ) -> Result<(), ConfigError> {
+        self.check_config()?;
+
+        log::trace!("slice reader: searching via multiline strategy");
+        let mut query_cursor = QueryCursor::new();
+        let query = &query_context.query;
+        let capture_index = query_context.capture_index;
+        let filter = &query_context.filter;
+        let tree: Cow<'_, Tree> = tree.map_or_else(
+            || {
+                Cow::Owned(
+                    slice
+                        .parse(&mut get_parser(query_context.language), None)
```

Comment:
This is the `Parseable` trait that I added for abstracting away the `ropey::Rope`-specific-ness (as far as "do we parse this as one big string vs in chunks")